include/usb/class/usb_audio.h

```diff
@@ -0,0 +1,282 @@
+ *	  Applicable for all devices.
+ *
+ * @warning Host may not use all of configured features.
+ *
+ * @param evt Pointer to an event to be parsed by the App.
+ *	      Pointer sturct is temporary and is valid only during the
+ *	      execution of this callback.
+ */
+typedef void (*usb_audio_feature_updated_cb_t)(struct device *dev,
+				const struct usb_audio_feature_unit_evt *evt);
+
+/**
+ * @brief Audio callbacks used to interact with audio devices by user App.
+ *
+ * usb_audio_ops structure contains all relevant callbacks to interact with
+ * USB Audio devices. Each of this callbacks is optional and may be left NULL.
+ * This will not break the stack but could make USB Audio device useless.
+ * Depending on the device some of those callbacks are necessary to make USB
+ * device work as expected sending/receiving data. For more information refer
+ * to callback documentation above.
+ */
+struct usb_audio_ops {
+	/* Callback called when data could be send */
+	usb_audio_data_request_cb_t data_request_cb;
+
+	/* Callback called when data were successfully written with sending
+	 * capable device. Applicable for headset and microphone. Unused for
+	 * headphones.
+	 */
+	usb_audio_data_completion_cb_t data_written_cb;
+
+	/* Callback called when data were successfully received by receive
+	 * capable device. Applicable for headset and headphones. Unused for
+	 * microphone.
+	 */
+	usb_audio_data_completion_cb_t data_received_cb;
+
+	/* Callback called when features were modified by the Host */
+	usb_audio_feature_updated_cb_t feature_update_cb;
+};
+
+/** @brief Get the frame size that is accepted by the Host.
+ *
+ * This function returns the frame size for Input Devices that is expected
+ * by the Host. Returned value rely on Input Device configuration:
+ * - number of channels
+ * - sampling frequency
+ * - sample resolution
+ * Device configuration is done via DT overlay.
+ *
+ * @param dev The Input device that is asked for frame size.
+ *
+ * @warning Do not use with OUT only devices (Headphones).
+ *	    For OUT only devices this function shall return 0.
+ */
+size_t usb_audio_get_in_frame_size(const struct device *dev);
+
+/**
+ * @brief Register the USB Audio device and make it useable.
+ *	  This must be called in order to make the device work
+ *	  and respond to all relevant requests.
+ *
+ * @param dev USB Audio device
+ * @param ops USB audio callback structure. Callback are used to
+ *	      inform the user about what is happening
+ */
+void usb_audio_register(struct device *dev,
+			const struct usb_audio_ops *ops);
+
+/**
+ * @brief Send data using USB Audio device
+ *
+ * @param dev    USB Audio device which will send the data
+ *		 over its ISO IN endpoint
+ * @param buffer Pointer to the buffer that should be send. User is
+ *		 responsible for managing the buffer for Input devices.
+ *		 In case of sending error user must decide if buffer
+ *		 should be dropped or retransmitted.
```

Comment:
``` Pointer to the buffer that should be send. User is responsible for managing the buffer for Input devices. In case of sending error user must decide if the buffer should be dropped or retransmitted. Afther the buffer was sent successfully it is passed to the data_written_cb callback if the application uses one or automatically freed otherwse. User must provide proper net_buf chunk especially when it comes to its size. This information can be obtained using usb_audio_get_in_frame_size() API function. ```